Fix macro validity checks

PappasBrent opened the pull request:
- Refactor the definitions for `shouldSkipMacroDefinition()` and `shouldSkipMacroInvocation()` to be easier to read and debug.
- Fix the call to `shouldSkipMacroInvocation()` by passing the spelling location of macro expansions' begin locations to handle nested expansions correctly.
- Change the expected output of tests involving nested macro expansions to reflect these changes.

Closes #48

PappasBrent committed:
Use ifs for macro checks instead of a bool expr

Implement the "should skip macro" checks using if statements instead of
a single boolean expression so that the code is more legible and easier
to insert debug statements into if need be.

PappasBrent committed:
Use llvm instead of llvm-17 for portability

PappasBrent committed:
Pass spelling loc to shouldSkipMacroInvocation()

When processing a macro expansion, pass its spelling location to
shouldSkipMacroInvocation() instead of the location given by Clang's
MacroExpands() hook to properly handle nested macro invocations.

PappasBrent committed:
Remove getSpellingRange()

Remove getSpellingRange() and just compute the end spelling location
separately to avoid recomputing the begin spelling location for macro
invocations twice.

PappasBrent committed:
Fix tests involving nested invocations

Change the expected output for tests containing nested invocations to
contain nested expansions that were prreviously considered invalid and
not reported.

SilverMight commented on lib/MakiFlags.cc:
```diff
@@ -7,21 +7,47 @@ namespace maki {
+        return true;
+    }
+    if (!Flags.ProcessMacrosAtInvalidLocations) {
+        [[maybe_unused]] auto [Valid, LocationOrError] =
```
Very minor but I think it'd be more appropriate to use `std::tie` and `std::ignore` instead of `[[maybe_unused]]` attribute especially because LocationOrError is always unused

PappasBrent committed:
Replace [[maybe_unused]] with std::ignore

Replace [[maybe_unused]] with std::ignore to better convey the fact the
LocationOrError value is unused.
